.github/workflows/ci.yaml

```diff
@@ -15,21 +15,82 @@ env:
   NODE_OPTIONS: --max_old_space_size=6144
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+permissions: {}
+
 jobs:
+  dedupe:
+    name: Deduplicate dependencies
+    # Skip unless this is a dependabot pull request
+    if: |
+      github.actor == 'dependabot[bot]' &&
+      startsWith(github.head_ref, 'dependabot/npm_and_yarn/')
```

Comment:
This will create constant "skipped" statuses in every pull request and commit. That is not pretty or preferred IMHO.

Maybe we should split it out into its own workflow and only trigger pull requests for these branches only.

Reply:
Simply cannot do that without using a PAT or other authenticated script to re-trigger the CI workflow per my original comment.

Reply:
I think I addressed all your concerns except this one, which just can't be helped unless I'm really missing something.  If it's any consolation, there are a number of other common skips in the HA repos, including the design preview in the frontend.

Reply:
Those need to be visible on all in case they are used because we can not detect when they are needed. Here, we can and can adjust the trigger to only trigger when needed.
Like this in a separate workflow file:
```yaml
on:
  push:
    branches: 
      - 'dependabot/npm_and_yarn/**'
```
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags

Reply:
Per my other comments, can't be a separate workflow unless it authenticates using something other than the `GITHUB_TOKEN` because it would never re-trigger the CI workflow (and you'd always waste resources because CI won't wait for the separate workflow).

Reply:
> Per my other comments, can't be a separate workflow unless

That sounds like an incorrect statement... it would run with the exact same permissions (in the way proposed by @ludeeus).

Reply:
https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

Reply:
Its not triggered by a workflow, its triggered by a push to a spesific branch.

Reply:
The issue is re-triggering the CI.  If the deduplication is a separate workflow and commits with `GITHUB_TOKEN`, the CI workflow will remain failed because no event is triggered.

I get this isn't pretty.  I don't like it either.  I'm just trying to automate within the constraints of GitHub and dependabot.  If you give me a PAT or GitHub app to authenticate with instead, the problem goes away and it can be a separate workflow triggered via push or even manually with a label.